bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```diff
@@ -0,0 +1,684 @@
+                                                HeliosEasyControlsHandler.class.getSimpleName(), errorMsg);
+                                        lock.release();
+                                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                                errorMsg);
+                                    });
+                        }, failureInfo -> {
+                            String errorMsg = failureInfo.getCause().getMessage();
+                            logger.warn("{} encountered error writing to device: {}",
+                                    HeliosEasyControlsHandler.class.getSimpleName(), errorMsg);
+                            lock.release();
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorMsg);
+
+                        });
+            }
+        } else {
+            if (this.comms == null) {
+                this.handleError("Modbus communication interface is unavailable.",
+                        ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+            if (this.vMap == null) {
+                this.handleError("Variable definition is unavailable.", ThingStatusDetail.CONFIGURATION_ERROR);
+            }
+        }
+    }
+
+    // TODO: set time incl. timezone, update time channels based on timezone read from device
+
+    /**
+     * Formats the provided date to a string in the device's configured date format
+     *
+     * @param variableName the variable name
+     * @param date the date to be formatted
+     * @return a string in the device's configured date format
+     */
+    private String formatDate(String variableName, ZonedDateTime date) {
+        String y = Integer.toString(date.getYear());
+        String m = Integer.toString(date.getMonthValue());
+        if (m.length() == 1) {
+            m = "0" + m;
+        }
+        String d = Integer.toString(date.getDayOfMonth());
+        if (d.length() == 1) {
+            d = "0" + d;
+        }
+        if (variableName.equals(HeliosEasyControlsBindingConstants.DATE)) { // fixed format for writing the system date
+            return d + "." + m + "." + y;
+        } else {
+            switch (this.dateFormat) {
+                case 0: // dd.mm.yyyy
+                    return d + "." + m + "." + y;
+                case 1: // mm.dd.yyyy
+                    return m + "." + d + "." + y;
+                case 2: // yyyy.mm.dd
+                    return y + "." + m + "." + d;
+                default:
+                    return d + "." + m + "." + y;
+            }
+        }
+    }
+
+    /**
+     * Returns a DateTimeType object based on the provided String and the device's configured date format
+     *
+     * @param date The date string read from the device
+     * @return A DateTimeType object representing the date or time specified
+     */
+    private DateTimeType toDateTime(String date) {

```

Comment:
Isn't the date format fixed in the protocol?

Reply:
The date format can be configured to 3 different formats. I use this method to create a DateTimeType I can actually work with. As mentioned above, I do intend to make the handling of system date and time a bit more convenient...